pkg/dockerregistry/server/configuration/configuration.go

```diff
@@ -151,3 +249,165 @@ func pushEnv(environ []envVar) error {
+		if authType != middlewareName {
+			log.Errorf("wrong configuration detected, registry should use openshift auth controller: %v", authType)
+			config.Auth = make(configuration.Auth)
+			config.Auth[middlewareName] = make(configuration.Parameters)
+		}
+	*/
+}
+
+func getServerAddr(options configuration.Parameters) (registryAddr string, err error) {
+	// TODO: Deprecate this environment variable.
+	registryAddr = os.Getenv(DockerRegistryURLEnvVar)
+	if len(registryAddr) == 0 {
+		registryAddr = os.Getenv(OpenShiftDefaultRegistryEnvVar)
+	} else {
+		log.Infof("DEPRECATED: %q is deprecated, use the %q instead", DockerRegistryURLEnvVar, OpenShiftDefaultRegistryEnvVar)
+	}
+	if len(registryAddr) == 0 {
+		registryAddr, err = getStringOption(DockerRegistryURLEnvVarOption, "dockerregistryurl", registryAddr, options)
+		if err != nil {
+			return
+		}
+	}
+
+	// TODO: This is a fallback to assuming there is a service named 'docker-registry'. This
+	// might change in the future and we should make this configurable.
+	if len(registryAddr) == 0 {
+		if len(os.Getenv("DOCKER_REGISTRY_SERVICE_HOST")) > 0 && len(os.Getenv("DOCKER_REGISTRY_SERVICE_PORT")) > 0 {
+			registryAddr = os.Getenv("DOCKER_REGISTRY_SERVICE_HOST") + ":" + os.Getenv("DOCKER_REGISTRY_SERVICE_PORT")
+		} else {
+			err = fmt.Errorf("%s variable must be set when running outside of Kubernetes cluster", DockerRegistryURLEnvVar)
+			return
+		}
+	}
+
+	return
+}
+
+func migrateMiddleware(dockercfg *configuration.Configuration, cfg *Configuration) (err error) {
+	var repoMiddleware *configuration.Middleware
+	for _, middleware := range dockercfg.Middleware["repository"] {
+		if middleware.Name == middlewareName {
+			repoMiddleware = &middleware
+			break
+		}
+	}
+	if repoMiddleware == nil {
+		repoMiddleware = &configuration.Middleware{
+			Name:    middlewareName,
+			Options: make(configuration.Parameters),
+		}
+	}
+
+	if cfg.Auth == nil {
+		cfg.Auth = &Auth{}
+		cfg.Auth.Realm, err = getStringOption("", RealmKey, "origin", dockercfg.Auth.Parameters())
+		if err != nil {
+			return
+		}
+		cfg.Auth.TokenRealm, err = getStringOption("", TokenRealmKey, "", dockercfg.Auth.Parameters())
+		if err != nil {
+			return
+		}
+	}
+	if cfg.Audit == nil {
+		cfg.Audit = &Audit{}
+		authParameters := dockercfg.Auth.Parameters()
+		if audit, ok := authParameters["audit"]; ok {
+			auditOptions := make(map[string]interface{})
+
+			for k, v := range audit.(map[interface{}]interface{}) {
+				if s, ok := k.(string); ok {
+					auditOptions[s] = v
+				}
+			}
+
+			cfg.Audit.Enabled, err = getBoolOption("", "enabled", false, auditOptions)
+			if err != nil {
+				return
+			}
+		}
+	}
+	if cfg.Server == nil {
+		cfg.Server = &Server{}
+		cfg.Server.Addr, err = getServerAddr(repoMiddleware.Options)
+		if err != nil {
+			return
+		}
+	}
+	if cfg.Quota == nil {
+		cfg.Quota = &Quota{}
+		cfg.Quota.Enabled, err = getBoolOption(EnforceQuotaEnvVar, "enforcequota", false, repoMiddleware.Options)
```

Comment:
the godoc for EnforceQuotaEnvVar says it takes precedence over what's in the config, but that's not what appears to be happening here?

Reply:
@bparees Good catch! Fixed.